### PR TITLE
Added skipFailingTests option from maven plugin

### PR DESCRIPTION
**Pull request by nicerloop**
Added an option to ignore failing tests from coverage, activated from maven plugin

### CODE REVIEW

File: pitest-entry/src/main/java/org/pitest/coverage/execute/DefaultCoverageGenerator.java [hunk 1]
```diff
@@ -179,7 +179,9 @@ private void gatherCoverageData(final Collection<ClassInfo> tests,
 
       @Override
       public void apply(final CoverageResult cr) {
-        coverage.calculateClassCoverage(cr);
+        if (cr.isGreenTest() || !coverageOptions.getPitConfig().skipFailingTests()) {
```
Comment (pedrorijo91) [inline pitest-entry/src/main/java/org/pitest/coverage/execute/DefaultCoverageGenerator.java]: should we at least log that we are skipping coverage computation due to flag+failed tests?

File: pitest-maven/src/main/java/org/pitest/maven/SurefireConfigConverter.java [hunk 2]
```diff
@@ -79,4 +80,11 @@ private void convertExcludes(ReportOptions option, Xpp3Dom configuration) {
     return Collections.emptyList();
   }
 
+  private void convertTestFailureIgnore(ReportOptions option, Xpp3Dom configuration) {
+    List<String> testFailureIgnoreList = extractStrings("testFailureIgnore", configuration);
+    if (!testFailureIgnoreList.isEmpty()) {
```
Comment (pedrorijo91) [inline pitest-maven/src/main/java/org/pitest/maven/SurefireConfigConverter.java]: what happens if the list contains more than 1 element? is it ignored? should we log?

Comment (nicerloop) [reply pitest-maven/src/main/java/org/pitest/maven/SurefireConfigConverter.java]: I could just skip the list conversion and directly extract the value.

File: pitest-entry/src/main/java/org/pitest/coverage/execute/DefaultCoverageGenerator.java (hunk 1 above)
Comment (nicerloop) [reply pitest-entry/src/main/java/org/pitest/coverage/execute/DefaultCoverageGenerator.java]: I actually don't know the strategy around logs in this project, just hacked for my immediate need to skip failing tests

Comment (nicerloop) [reply pitest-entry/src/main/java/org/pitest/coverage/execute/DefaultCoverageGenerator.java]: Where in the code can I see the recommended pattern to add logging?

**Comment by nicerloop** [comment]
Is this evolution in the line of the project? May it be integrated? What would be required for that?

**Comment by hcoles** [comment]
@nicerloop sorry for the slow response, unfortunately I'm a little short of time these days.

Skipping failing tests has been requested as a feature a few times now, I've always been reluctant to implement it as it adds one more variable to consider when people raise an issue because they believe there is a falsely killed mutant. If a failing test is not properly excluded from running it will appear to kill all mutants. 

Not sure if this is still a rational concern or not: I'll take a look at the PR when I get chance.

**Comment by dormaayan** [comment]
Thanks for implementing this! It seems useful!
Is there a plan to add this feature to the upcoming release of Pitest?

**Comment by maxgabut** [comment]
From the merge date, it's included in version 1.4.4.

It's lacking documentation at that point, but adding `<skipFailingTests>true</skipFailingTests>` in the plugin configuration in you pom should do the trick (not tested, but from the code it should work)

**Pull request merged**